core/src/main/java/io/aiven/kafka/tieredstorage/chunkmanager/cache/ChunkCache.java

```diff
@@ -138,4 +142,29 @@ protected AsyncCache<ChunkKey, T> buildCache(final ChunkCacheConfig config) {
         statsCounter.registerSizeMetric(cache.synchronous()::estimatedSize);
         return cache;
     }
+
+    public void startPrefetching(final String segmentKey,
```

Comment:
Wondering if this could be refactor to receive a set of chunk ids to prefetch instead of calculate based on range, and move the calculation to the FetchChunkEnumeration constructor where start and end chunk id are already calculated.

If we switch the config to be by number of chunks instead of bytes, then we can do the prefetching directly here instead of the initial prep. wdyt?